pkg/odo/genericclioptions/runnable.go

```diff
@@ -3,21 +3,39 @@ package genericclioptions
 import (
 	"flag"
 	"fmt"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"k8s.io/klog"
 	"os"
+	"time"
 
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/odo/util"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
+var (
+	segmentClient *segment.Client
+	//ctx context.Context
+)
+
 type Runnable interface {
 	Complete(name string, cmd *cobra.Command, args []string) error
 	Validate() error
 	Run() error
 }
 
 func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
+	//ctx = telemetry.NewContext(context.Background())
+	startTime := time.Now()
+	cfg, _ := preference.New()
```

Comment:
I would like you to consider the testablity of you code as well, assume you had to test this function `GenericRun` while you mock this preference object, could you be able to do that?
I would suggest - only pass the preference value you care about and not the whole object.

Reply:
>I would like you to consider the testablity of you code as well, assume you had to test this function GenericRun while you mock this preference object, could you be able to do that?

I am still not sure I understand what you mean here.

>I would suggest - only pass the preference value you care about and not the whole object.

I care about `ConsentTelemetry` in particular, but in case we want to pass other preference values to Segment like CRC has done [here](https://github.com/code-ready/crc/blob/master/pkg/crc/segment/segment.go#L104), it would be easier if we had the preference object.

Reply:
> I would like you to consider the testablity of you code as well, assume you had to test this function GenericRun while you mock this preference object, could you be able to do that?

I think I would be able to write the test. `preference.New()` returns the preference object as set in the preference.yaml file. If I wanted to mock, I would first create a separate file, set the env var `GlobalConfigEnvName` to use this new file, and then run a command as I normally would. I think [this test](https://github.com/openshift/odo/blob/master/pkg/preference/preference_test.go#L16) is a good example of doing this.

Correct me if I misunderstood your concern.